### PR TITLE
Adding `GetFunctionContext` extension method on `TaskOrchestrationContext`

### PR DESCRIPTION
**Pull request by kshyju**
# Summary

Adding `GetFunctionContext` extension method on `TaskOrchestrationContext`. This returns the `FunctionsContext` instance if the orchestrationcontext is running in Functions hosting context.

## Why is this change needed?
We need to retrieve a registered dependency from the DI container. This can be done via the `InstanceServices` property on FunctionContext. The `AddOrchestratorFunc` is used to register the orchestration to support scenarios where we override the function executor with our custom executor. In those cases, the `TaskOrchestrationContext` parameter fails to bind when used as a function input parameter (because that binding happens in the [DurableFunctionExecutor](https://github.com/Azure/azure-functions-durable-extension/blob/9eec0855d6ae0a57230bca9df0e86d73a2e71170/src/Worker.Extensions.DurableTask/Execution/DurableFunctionExecutor.Orchestration.cs#L24-L39)), particularly for Microsoft Agent Framework support scenarios.
```
functionApplicationBuilder.ConfigureDurableWorker().AddTasks(tasks =>
{
    tasks.AddOrchestratorFunc<string, string>(
        "MyCustomOrchestration",
        async (orchestrationContext, request) =>
        {
            FunctionContext functionContext = orchestrationContext.GetFunctionContext()
                ?? throw new InvalidOperationException("FunctionContext is not available!.");

            var runner = functionContext.InstanceServices.GetRequiredService<DurableWorkflowRunner>();
            ILogger logger = orchestrationContext.CreateReplaySafeLogger("MyCustomOrchestration");

            return await runner.RunWorkflowOrchestrationAsync(orchestrationContext, request, logger);
        });
});
``` 

## Issues / work items
n/a

---

# Project checklist
- [ ] Documentation changes are not required
  - [ ] Otherwise: Documentation PR is ready to merge and referenced in `pending_docs.md`
- [ ] Release notes are not required for the next release
  - [x] Otherwise: Notes added to `release_notes.md`
- [ ] Backport is not required
  - [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
- [x] All required tests have been added/updated (unit tests, E2E tests)
- [x] No extra work is required to be leveraged by OutOfProc SDKs
  - [ ] Otherwise: Work tracked here: #issue_or_pr_in_each_sdk
- [ ] No change to the version of the `WebJobs.Extensions.DurableTask` package
  - [x] Otherwise: Major/minor updates are reflected in `/src/Worker.Extensions.DurableTask/AssemblyInfo.cs`
- [x] No EventIds were added to `EventSource` logs
  - [ ] Otherwise: Ensure EventIds are within the supported range in the existing Windows infrastructure (validate via deployed telemetry). If needed, extend the range via a PR such as https://msazure.visualstudio.com/One/_git/AAPT-Antares-Websites/pullrequest/7463263?_a=files
- [ ] This change should be added to the `v2.x` branch
  - [ ] Otherwise: This change applies exclusively to `WebJobs.Extensions.DurableTask` v3.x and will be retained only in the `dev` and `main` branches
- [ ] Breaking change?
  - [ ] If yes:
    - Impact:
    - Migration guidance:
---

# AI-assisted code disclosure (required)
## Was an AI tool used? (select one)
- [ ] No
- [x] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [ ] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s):
- AI-assisted areas/files:
- What you changed after AI output:

AI verification (required if AI was used):
- [x] I understand the code and can explain it
- [x] I verified referenced APIs/types exist and are correct
- [x] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [x] I reviewed concurrency/async behavior
- [x] I checked for unintended breaking or behavior changes

---

# Testing
## Automated tests
- Result: Passed / Failed (link logs if failed)

## Manual validation (only if runtime/behavior changed)
- Environment (OS, .NET version, components):
- Steps + observed results:
  1.
  2.
  3.
- Evidence (optional):

---

# Notes for reviewers
- N/A

### CODE REVIEW

File: src/Worker.Extensions.DurableTask/Worker.Extensions.DurableTask.csproj [hunk 1]
```diff
@@ -29,7 +29,7 @@
     <AssemblyOriginatorKeyFile>..\..\sign.snk</AssemblyOriginatorKeyFile>
 
     <!-- Version information -->
-    <VersionPrefix>1.13.1</VersionPrefix>
+    <VersionPrefix>1.13.2</VersionPrefix>
```
Comment (cgillum) [inline src/Worker.Extensions.DurableTask/Worker.Extensions.DurableTask.csproj]: Since this is a new public API, should we bump the minor version instead?

Comment (kshyju) [reply src/Worker.Extensions.DurableTask/Worker.Extensions.DurableTask.csproj]: Bumped minor version instead of patch, in the new iteration pushed.

**Review by cgillum — APPROVED** [review]
